pandas/_libs/groupby_helper.pxi.in

```diff
@@ -725,24 +725,37 @@ def group_min_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
                     out[i, j] = minx[i, j]
 
 
+
+{{endfor}}
+
+
+ctypedef fused groupby_t:
+    float64_t
+    float32_t
+    int64_t
+
+
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def group_cummin_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
-                          ndarray[{{dest_type2}}, ndim=2] values,
-                          ndarray[int64_t] labels,
-                          bint is_datetimelike):
+def group_cummin(ndarray[groupby_t, ndim=2] out,
+                 ndarray[groupby_t, ndim=2] values,
+                 ndarray[int64_t] labels,
+                 bint is_datetimelike):
     """
     Only transforms on axis=0
     """
     cdef:
         Py_ssize_t i, j, N, K, size
-        {{dest_type2}} val, mval
-        ndarray[{{dest_type2}}, ndim=2] accum
+        groupby_t val, mval
+        ndarray[groupby_t, ndim=2] accum
         int64_t lab
 
     N, K = (<object> values).shape
     accum = np.empty_like(values)
-    accum.fill({{inf_val}})
+    if groupby_t is int64_t:
```

Comment:
can u make this more generic ? she what if we expand this to other int types?

Reply:
this is a small cost of compile time (actually maybe nothing as cython is pretty smart). but i suppose can handle later.

Reply:
Fair enough.  Easy to implement if/when its actually needed.

Reply:
see some comments above